### PR TITLE
Sentry: use processor to set GatewayTimeout errors to :warning

### PR DESCRIPTION
**Pull request by jholton**
## Description of change
create a Sentry processor to set log level to :warning for `GatewayTimeout` errors. so we don't have to worry about manually logging it everywhere it can occur (`ApplicationController` any background job, and any line of code that `rescues` `StandardError` and logs it.)

## Testing done
specs

## Testing planned
manual testing in dev/staging

#### Unique to this PR

- [x] create a Sentry processor to set log level to :warning for `GatewayTimeout` errors
- [x] eliminate `SentryIgnoredGatewayTimeout` exception class and `rescue_timeout` middlewares

#### Applies to all PRs

- [x] Appropriate logging
- [ ] Swagger docs have been updated, if applicable
- [ ] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: lib/sentry/processor/log_as_warning.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      def process(data)
+        process_if_symbol_keys(data) if data[:exception]
+        process_if_string_keys(data) if data['exception']
+        data
+      end
+
+      private
+
+      def process_if_symbol_keys(data)
+        data[:level] = 30 if data[:exception][:values].last[:type] == 'Common::Exceptions::GatewayTimeout'
```
Comment (lihanli) [inline lib/sentry/processor/log_as_warning.rb]: can you make the log level and exception name constants?

Comment (jholton) [reply lib/sentry/processor/log_as_warning.rb]: done.

**Comment by saneshark** [comment]
WOW, this looks great. I had no idea about these processors! I think we could leverage something like this to get the messages that Sentry reports for BackendServiceException to be grouped better / differently... right now they are all grouped together as one common BackendServiceException group

**Comment by jholton** [comment]
relevant issue:  https://github.com/department-of-veterans-affairs/vets.gov-team/issues/10315

**Pull request merged**